pkg/opensearch/client/client.go

```diff
@@ -54,7 +55,11 @@ func NewDatasourceHttpClient(ctx context.Context, ds *backend.DataSourceInstance
 		if settings.IsServerless {
 			httpClientOptions.SigV4.Service = "aoss"
 		}
-		httpClientOptions.Middlewares = append(httpClientOptions.Middlewares, sigV4Middleware())
+		authSettings, _ := awsds.ReadAuthSettingsFromContext(ctx)
```

Comment:
I think you have to use the deprecated ReadAuthSettings or it doesn't work on older versions of grafana

Reply:
Hm... we should probably un-deprecate that, in that case. Given how it works under the hood, maybe we should just remove the deprecation and make `ReadAuthSettingsFromContext` private?